### ISSUE TITLE
Doesn't install NPM

### ISSUE THREAD

**Issue by ddcado** [coreybutler/nvm-windows#766]
I use nvm to install node on windows as recommanded by microsoft.
After, npm wasn't install.
I saw that the npm mirror by default is "https://github.com/npm/cli/archive/" but this page doen't exist any more "err 404 not found"
I don't know what is the good uri....
I finally used node.js msi from the node js.com.
Kind regards.

**Comment by coreybutler** [comment]
Are you using an old version of NVM4W? This was fixed in version 1.1.7 (and above).

**Comment by ddcado** [comment]
1.1.
![Capture](https://user-images.githubusercontent.com/72383396/156200333-cf3cc86e-3b14-497c-a1d8-7cb6d7f103dc.PNG)
9

**Comment by tchari** [comment]
I have the same issue.
![image](https://user-images.githubusercontent.com/28321303/156210473-a6d71dea-b6b3-4247-aa86-f9f734dd03f4.png)

**Comment by ddcado** [comment]
You have to : 
nvm liste available
nvm install <16.14.0>
After try nvm liste
Et nom -v

**Comment by xuke-hat** [comment]
I have the same issue.

**Comment by WORMSS** [comment]
Stale because noone has fixed it, not because noone wants it fixed.

**Comment by thengoster** [comment]
@WORMSS The "Downloading npm" message does not appear because web.go added additional logic to the GetNodeJS function from v1.1.7 -> v1.1.9: https://github.com/coreybutler/nvm-windows/blob/f792f4ec5e3b4c5abe541002118e7c1bba1844ac/src/web/web.go#L142-L183

**Comment by WORMSS** [comment]
Yeah, just too many things are weird in 1.18 and higher. Things just not working and so on.. 
I've found it easier to just stay on 1.17. I've not seen a reason to upgrade yet. No functionality added, just bugs.

**Comment by baybars1223** [comment]
**EDIT**: Closing and reopening the command prompt terminal seems to have resolved the issue for me

It seems to be an issue of nvm not 'switching' rather than not installing. On `nvm 1.1.9`, `npm` is 'stuck' at `6.12.0` no matter which version I install or use. But if you navigate to `%APPDATA%\nvm\v16.14.2\node_modules\npm`, for example, the correct version is present.

Log of attempts to uninstall/reinstall:
```
C:\Users\XXX>nvm use 16.14.2
Now using node v16.14.2 (64-bit)

C:\Users\XXX>npm -v
6.12.0

C:\Users\XXX>nvm install latest
Downloading node.js version 17.9.0 (64-bit)...
Extracting...
Complete


Installation complete. If you want to use this version, type

nvm use 17.9.0

C:\Users\XXX>npm -v
6.12.0

C:\Users\XXX>nvm list

    17.9.0
  * 16.14.2 (Currently using 64-bit executable)
    12.13.1

C:\Users\XXX>nvm uninstall 17.9.0
Uninstalling node v17.9.0... done
C:\Users\XXX>nvm list

  * 16.14.2 (Currently using 64-bit executable)
    12.13.1

C:\Users\XXX>nvm install latest
Downloading node.js version 17.9.0 (64-bit)...
Extracting...
Complete


Installation complete. If you want to use this version, type

nvm use 17.9.0

C:\Users\XXX>nvm use 17.9.0
Now using node v17.9.0 (64-bit)

C:\Users\XXX>node -v
v17.9.0

C:\Users\XXX>npm -v
6.12.0

C:\Users\XXX>nvm uninstall 12.13.1
Uninstalling node v12.13.1... done
C:\Users\XXX>nvm uninstall 16.14.2
Uninstalling node v16.14.2... done
C:\Users\XXX>npm -v
6.12.0

C:\Users\XXX>nvm install 12.13.1
Downloading node.js version 12.13.1 (64-bit)...
Complete
Creating C:\Users\XXX\AppData\Roaming\nvm\temp

Downloading npm version 6.12.1... Complete
Installing npm v6.12.1...

Installation complete. If you want to use this version, type

nvm use 12.13.1

C:\Users\XXX>nvm use 12.13.1
Now using node v12.13.1 (64-bit)

C:\Users\XXX>node -v
v12.13.1

C:\Users\XXX>npm -v
6.12.0

C:\Users\XXX>nvm install 16.14.2
Downloading node.js version 16.14.2 (64-bit)...
Extracting...
Complete


Installation complete. If you want to use this version, type

nvm use 16.14.2

C:\Users\XXX>nvm use 16.14.2
Now using node v16.14.2 (64-bit)

C:\Users\XXX>node -v
v16.14.2

C:\Users\XXX>npm -v
6.12.0
```

**Comment by xujimu** [comment]
The problem remains unresolved

**Comment by douglasg14b** [comment]
This is still an issue, NPM is not installed...

on 1.1.9 no erorrs, just doesn't try and install PM at all.

On 1.1.8 it seems to try, and fail:

> Rollback failed. remove C:\Users\Douglas.Gaskell\AppData\Roaming\nvm\temp\npm-v6.14.17.zip: The process cannot access the file because it is being used by another process.
Could not download npm for node v14.19.2.
Please visit https://github.com/npm/cli/releases/tag/v6.14.17 to download npm.
It should be extracted to C:\Users\Douglas.Gaskell\AppData\Roaming\nvm\v14.19.2

Unfortunately there are no builds at https://github.com/npm/cli/releases/tag/v6.14.17 just source code bundles...

**Comment by douglasg14b** [comment]
Has anyone gotten this working on their device? If so, how?

**Comment by douglasg14b** [comment]
It's probably failing to install because the download fails, because this path: https://github.com/npm/cli/archive/v6.14.17.zip

Returns: `the given path has multiple possibilities: #<Git::Ref:0x00007f10c0794050>, #<Git::Ref:0x00007f10c0756548>`

**Comment by mrtawil** [comment]
> This is still an issue, NPM is not installed... Which kind of, completely, breaks things.
> 
> on 1.1.9 no erorrs, just doesn't try and install NPM at all.
> 
> On 1.1.8 it seems to try, and fail:
> 
> > Downloading npm version 6.14.17... Download failed. Rolling Back.
> > Rollback failed. remove C:\Users\Douglas.Gaskell\AppData\Roaming\nvm\temp\npm-v6.14.17.zip: The process cannot access the file because it is being used by another process.
> > Could not download npm for node v14.19.2.
> > Please visit https://github.com/npm/cli/releases/tag/v6.14.17 to download npm.
> > It should be extracted to C:\Users\Douglas.Gaskell\AppData\Roaming\nvm\v14.19.2
> 
> NVM will always say there is another process accessing the file, even after a fresh restart. I can freely go in and delete it.
> 
> Unfortunately there are no builds at https://github.com/npm/cli/releases/tag/v6.14.17 just source code bundles... If I extract that into the dir as instructed, it still does not work. If I copy the executables from `\bin` into the `\v14.19.2` directory, running `npm` just throws errors about being unable to find `'C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js'`
> 
> There is no npm executable in the install dir:
> 
> ![image](https://user-images.githubusercontent.com/1400380/167916369-e8fa22ff-b8e4-424d-8c89-ab0c27d8d2c2.png)

Same issue

**Comment by coreybutler** [comment]
`The process cannot access the file because it is being used by another process.` is caused by the OS locking a directory. This can be caused by any application using one of the executables in the directory (like a running process). This can also be caused by being in the directory that you're trying to change to/from on the terminal.

For example, if NVM4W is installed at `C:\nvm4w`, then Node 16.14.2 will be installed in `C:\nvm4w\v16.14.2`. If you open a terminal and cd to `C:\nvm4w\v16.14.2` and try to run `nvm use x.x.x`, Windows will lock the directory that's open in the terminal. This creates a conflict when trying to change the symlink. This isn't necessarily the case for everyone in this thread. I'm just pointing out the error message originates from the OS.

In the case of npm 6.14.2, it's just a bad link (as @douglasg14b pointed out). See #799.

**Comment by pedroyan** [comment]
Hi folks, 

I came across this issue and none of the other suggestions were working, such as [Enabling Developer Mode](https://github.com/coreybutler/nvm-windows/issues/580#issuecomment-1123815218) or [running NVM from a privileged prompt](https://github.com/coreybutler/nvm-windows/issues/803#issuecomment-1124251687).

![image](https://user-images.githubusercontent.com/11574849/168313590-ee7c0b1d-0134-4502-a79c-9096a28d1b01.png)

What fixed it for me was downgrading NVM to [1.1.7](https://github.com/coreybutler/nvm-windows/releases/tag/1.1.7). Hopefully this might also help you too.

**Comment by pedroyan** [comment]
Ok, just finished my experiment. The NPM Install problem begins when trying to install Node v14.19.2

```
PS C:\Users\pedro> nvm install 14.19.2
Downloading node.js version 14.19.2 (64-bit)...
Complete
Creating C:\Users\pedro\AppData\Roaming\nvm\temp

Downloading npm version 6.14.17... Download failed. Rolling Back.
ause it is being used by another process.
Could not download npm for node v14.19.2.
Please visit https://github.com/npm/cli/releases/tag/v6.14.17 to download npm.
It should be extracted to C:\Users\pedro\AppData\Roaming\nvm\v14.19.2
PS C:\Users\pedro> nvm uninstall 14.19.2
Uninstalling node v14.19.2... done
PS C:\Users\pedro> nvm install 14.19.2
Downloading node.js version 14.19.2 (64-bit)...
Complete
Creating C:\Users\pedro\AppData\Roaming\nvm\temp
Downloading npm version 6.14.17... Download failed. Rolling Back.
Rollback failed. remove C:\Users\pedro\AppData\Roaming\nvm\temp\npm-v6.14.17.zip: The process cannot access the file because it is being used by another process.
Could not download npm for node v14.19.2.
Please visit https://github.com/npm/cli/releases/tag/v6.14.17 to download npm.
It should be extracted to C:\Users\pedro\AppData\Roaming\nvm\v14.19.2
PS C:\Users\pedro> nvm install 14.18.3
Downloading node.js version 14.18.3 (64-bit)...
Complete
Creating C:\Users\pedro\AppData\Roaming\nvm\temp
Downloading npm version 6.14.15... Complete
Installing npm v6.14.15...

Installation complete. If you want to use this version, type

nvm use 14.18.3
PS C:\Users\pedro> nvm install 14.19.0
Downloading node.js version 14.19.0 (64-bit)...
Complete
Creating C:\Users\pedro\AppData\Roaming\nvm\temp

Downloading npm version 6.14.16... Complete
Installing npm v6.14.16...

Installation complete. If you want to use this version, type

nvm use 14.19.0
PS C:\Users\pedro> nvm install 14.19.1
Downloading node.js version 14.19.1 (64-bit)...
Complete
Creating C:\Users\pedro\AppData\Roaming\nvm\temp

Downloading npm version 6.14.16... Complete
Installing npm v6.14.16...

Installation complete. If you want to use this version, type

nvm use 14.19.1
PS C:\Users\pedro> nvm install 14.19.2
Downloading node.js version 14.19.2 (64-bit)...
Complete
Creating C:\Users\pedro\AppData\Roaming\nvm\temp

Downloading npm version 6.14.17... Download failed. Rolling Back.
Rollback failed. remove C:\Users\pedro\AppData\Roaming\nvm\temp\npm-v6.14.17.zip: The process cannot access the file because it is being used by another process.
Could not download npm for node v14.19.2.
Please visit https://github.com/npm/cli/releases/tag/v6.14.17 to download npm.
It should be extracted to C:\Users\pedro\AppData\Roaming\nvm\v14.19.2
PS C:\Users\pedro>

```

**Comment by Herr-Sepp** [comment]
I can confirm [pedroyan](https://github.com/pedroyan) observations.

14.19.1 works
14.19.2:
```
nvm install 14.19.2

Downloading node.js version 14.19.2 (64-bit)...
Complete
Downloading npm version 6.14.17... Download failed. Rolling Back.
Rollback failed. remove C:\Program Files\nvm\temp\npm-v6.14.17.zip: Der Prozess kann nicht auf die Datei zugreifen, da sie von einem anderen Prozess verwendet wird.
Could not download npm for node v14.19.2.
Please visit https://github.com/npm/cli/releases/tag/v6.14.17 to download npm.
It should be extracted to C:\Program Files\nvm\v14.19.
```

**Comment by yoavain** [comment]
Problem is that the URL `https://github.com/npm/cli/archive/v6.14.7.zip` results in status code 302, which redirects to `https://codeload.github.com/npm/cli/zip/refs/tags/v6.14.7` (see location header).
The root cause is that the `Download` function does not follow this redirect. It fails on any status code other than 200.

**Comment by yoavain** [comment]
The second problem is that this tool is written in Go, and most of the developers looking for a fix are NodeJS developers.
Unfortunately, I currently don't have the time to learn Go to offer help.

**Comment by Kanyan88** [comment]
you can check    that NPM is installed  with the version you chose to install  from this directory `C:\Users\%USERNAME%\AppData\Roaming\nvm`  
the directories with `v` prefix are the install versions  you can check   npm exsist inside npm modules   
for people who installed for the first time  you should :
run cmd as Admin then list versions with `nvm list` then `nvm use __version_installed__`  otherwise nvm is will not be using any installation you downloaded

**Comment by coreybutler** [comment]
The issue was the URL. When a Github project has a branch and a tag with the same name, Github returns an HTTP status `300`, _not a redirect_ . If NVM4W encounters a status like this, it will now output the results to screen:

<img width="721" alt="image" src="https://user-images.githubusercontent.com/770982/170831224-4ced728d-1601-4786-903c-1d34c4143502.png">

There is also a specific error handler to work around this particular version of npm, since it appears to be the only version affected by this problem. I am not fond of these kinds of hacks, because this is ultimately an upstream issue.

**Comment by ArclightRSA** [comment]
Okay I am not sure if it has been mentioned, but I have a workaround for now...

`
Rename the two folders npm and npm-cache in C:\Users\MyUsername\AppData\Roaming to anything you desire, this will induce a rollback function.
`

My experience

I found that C:\Users\MyUsername\AppData\Roaming has two folders npm and npm-cache. I renamed them to npm.1 and npm-cache,1. This gives me access to npm again. What I am suspecting and correct me if I am wrong the roaming folder is where my global settings are stored as I noticed when I rename those folders it then looks at my Program Files Node folder.

I have multiple projects running atm that requires met to switch between Node 10.16.0, 10.18.0 and 16 hence why I would like to just understand why this work around is working.  When I switch between 10.18.0 and 16 and 18 I can use npm but when I switch back to 10.16 I had to redo the rename workaround.

I hope this helps someone in dire need like it helped me.

**Comment by coreybutler** [comment]
For those finding this issue for npm 6.14.17, there was an issue upstream in the npm github reo that has been fixed: https://github.com/npm/cli/issues/4890#issuecomment-1141577228.

**Comment by Viktor1993js** [comment]
Hi all. The following helped me. I needed version 14.16.1, but npm didn't load.
I wrote (nvm off), then downloaded https://nodejs.org/en/blog/release/v14.16.1 installer and installed it, after that I went into the folder with installed node.js  and copied all files to the folder nvm with version 14.16.1 which we need to create in advance before disabling. After that, I deleted node.js which I installed with the help of installer and prescribed nvm on, then switched to my version 14.16.1

**Comment by bengabp** [comment]
and i have thesame issue now

**Comment by WORMSS** [comment]
> and i have thesame issue now

This has not been a issue for many years. If you have all the latest version's of NVM, it would be worth creating a new issue with all the relevant information such as OS/NVM version/paths trying to use/etc

**Comment by ceres7n** [comment]
> You have to : nvm list available nvm install 16.14.0 After try nvm list Et nvm -v

worked, but makes no sense.
